trio/_abc.py

```diff
@@ -186,7 +209,12 @@ class SocketFactory(metaclass=ABCMeta):
     """
 
     @abstractmethod
-    def socket(self, family=None, type=None, proto=None):
+    def socket(
+        self,
+        family: socket.AddressFamily | int = socket.AF_INET,
+        type: socket.SocketKind | int = socket.SOCK_STREAM,
+        proto: int = 0,
+    ) -> _SocketType:
```

Comment:
Meh, I don't really feel comfortable changing away from the `=None` without knowing why exactly they're here in the first place. Additonally, in typeshed this is how a similar function is defined:

```py
if sys.platform == "win32":
    def socketpair(family: int = ..., type: int = ..., proto: int = 0) -> tuple[socket, socket]: ...

else:
    def socketpair(
        family: int | AddressFamily | None = None, type: SocketType | int = ..., proto: int = 0
    ) -> tuple[socket, socket]: ...
```

Reply:
Actually figuring out the value of the defaults required a bit of digging, and it took me a bit to be certain enough that they were static and not dynamic or something, given that pretty much all default values in typeshed is `= ...`. I wouldn't have set the default values to anything but `...` if not for `typeright --verifytypes` complaining when default values aren't explicit.

https://github.com/python-trio/trio/blob/315dbed5abf0663b11522022e9c7abbc1df54dc8/trio/_socket.py#L270-L289
does have explicit logic for `None` though, so reverting the default value to `None` seems eminently plausible.